middleware/SessionAuthMiddleware.php

```diff
@@ -44,8 +47,8 @@ public function __invoke(Request $request, RequestHandler $handler): Response
 			if ((!isset($_COOKIE[$this->SessionCookieName]) || !$sessionService->IsValidSession($_COOKIE[$this->SessionCookieName])) && $routeName !== 'login')
 			{
 				define('GROCY_AUTHENTICATED', false);
-				$response = new \Slim\Psr7\Response(); // No content when unauthorized
-				$response = $response->withHeader('Location', $this->AppContainer->get('UrlManager')->ConstructUrl('/login'));
+				$response = $this->responseFactory->createResponse();
+				return $response->withHeader('Location', $this->AppContainer->get('UrlManager')->ConstructUrl('/login'));
```

Comment:
@berrnd Minor; please also note the early `return` statement here for slight additional safety.